### DISCUSSION TITLE
OpenDTU won´t talk to home assistant via MQTT

### DISCUSSION THREAD

**Discussion by DerKlotz74** [tbnobody/OpenDTU#1795] [Q&A]
Hi folks, 
is it necessary to have a Hoymiles to pair the OpenDTU with Homeassistant via MQTT? My Hoymiles won't arrive for a few days and I'm trying to establish a connection. 
With all tips I always have the message in Homeassistant " 
New connection from 192.168.10.46:62766 on port 1883. 2024-02-26 18:49:49: Client OpenDTU-16623084 disconnected, not authorized."

I access my Raspi with HA via Windows. Under Mosquitto Broker Options I have entered the network address of the Pi or do I have to enter core-mosquitto rei? OpenDTU also runs under the Raspi address, user and password match.
I hope someone here can help me.

**Comment by MetaChuh** [comment]
hi @DerKlotz74 

> My Hoymiles won't arrive for a few days and I'm trying to establish a connection.

no, you can not establish a connection to your hoymiles until it arrives.

your hoymiles inverter has to be connected to your solar panels, has to be within your local radio range of about 100m, and you have to add its serial number to the dtu inverter list.

before that, it is impossible to connect to your inverter, due to distance between you and your inverter, until it reaches your home, and possibly not having added your inverter's serial number to opendtu, unless your seller has provided you with the inverter serial number, before shipping it.

hope this helps & greetings,
metachuh

**Comment by DerKlotz74** [comment]
Da habe ich mich mißverständlich ausgedrückt. Ich wollte nur, dass der OpenDTU mit Homeassistant auch ohne Wechselrichter kommuniziert. Gestern habe ich die Kommunikation endlich hinbekommen. Warum das plötzlich nach 3 Tagen erfolgloser Suche funktioniert hat weiß ich nicht. 

Danke

**Reply by foxfossil** [reply]
Dann weisst du aber schon mehr als ich und kannst mir vielleicht helfen:
Wie kriige ich den openDRU überhaupt dazu, sich beim HomeAssistant zu melden?

**Comment by DerKlotz74** [comment]
Moin,
ich habe das ganze auf meinem Laptop mit WIn11 und Chrome auf meinem Raspberry Pi4 eingerichtet. Wenn du direkt über deinen Pi installierst sollen teilweise andere Server EInstellungen wichtig sein habe ich heraus gelesen.

1. Mosquitto Broker in HA unter Addons installieren. EInstellungen können alle auf Default gelassen werden. MQTT default Port ist 1883. Diesen für weitere Schritte merken. MQTT starten und vorsichtshalber HA neu starten.
2. In HA unter Personen -> Neuen Benutzer erstellen. In den Anleitungen wird häufig mqtt-user erstellt. Passwort setzen (für weitere Schritte merken). Schieberegler Anmelden im lokalen Netzwerk aktivieren.
3. Die Netzwerkadresse deines Pi´s herausfinden
4. Unter HA -> Geräte Dienste -> Integration hinzufügen -> MQTT eingeben und auswählen. Konfiguration von MQTT anpassen. Serveradresse deines Pi´s, Port:1883, Benutzername:mqtt-user und Passwort vergeben.
5. Nun gehts zum OpenDTU. Dort unter MQTT die gleichen EInstellungen wie bei Schritt 4. Unter Baistopic habe ich solar/ entfernt und für meinen Hoymiles hms800/ gewählt. War auch irgendwo ein Tipp.
6. Jetzt OpenDTU und HA vorsichtshalber neu starten und mindestens 5 Minuten warten.

Du könntest dir noch unter HA ->Addons MQTT Explorer installieren. Dort stellst du eine neue Verbindung her mit der Netzwerkadresse des Pi und die anderen Daten wie unter 4. Jetzt Connect drücken und wenn alles geklappt hat, dann sollte dort dein Basistopic von 5. auftauchen.

Ich habe Node-Red installiert und vermutlich deswegen Probleme mit meiner Verbindung gehabt. Falls du das auch hast... In Node-Red eine MQTT Node in den Workflow ziehen, doppelklicken und dort unter Server -> MQTT bearbeiten (Bleistiftsymbol) -> Server die Adresse deines Pi, jetzt Reiter Sicherheit wählen und dort Benutzername und Passwort eingeben. Jetzt auf aktualisieren klicken. Rechts oben auf Übernahme/Deploy und Bestätigen. Nun vorsichtshalber HA neu starten und wieder warten... 

Viel Erfolg

**Comment by foxfossil** [comment]
Herzlichen Dank!Jetzt habe ich vielleicht die Richtung.Ich habe keinen Pi im Einsatz! sondern einen mini-PC für 23€ + 256GB SSD- openHAOS drauf und an der Fritz!Box angemeldet.Mal sehen, wie ich mit Deinen Angaben weiterkomme ;)>--Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.

 
  
  
 
 Am 03.03.24, 11:21 schrieb DerKlotz74 ***@***.***>:

   
  Moin, ich habe das ganze auf meinem Laptop mit WIn11 und Chrome auf meinem Raspberry Pi4 eingerichtet. Wenn du direkt über deinen Pi installierst sollen teilweise andere Server EInstellungen wichtig sein habe ich heraus gelesen. 
   
   Mosquitto Broker in HA unter Addons installieren. EInstellungen können alle auf Default gelassen werden. MQTT default Port ist 1883. Diesen für weitere Schritte merken. MQTT starten und vorsichtshalber HA neu starten. 
   In HA unter Personen -> Neuen Benutzer erstellen. In den Anleitungen wird häufig mqtt-user erstellt. Passwort setzen (für weitere Schritte merken). Schieberegler Anmelden im lokalen Netzwerk aktivieren. 
   Die Netzwerkadresse deines Pi´s herausfinden 
   Unter HA -> Geräte Dienste -> Integration hinzufügen -> MQTT eingeben und auswählen. Konfiguration von MQTT anpassen. Serveradresse deines Pi´s, Port:1883, Benutzername:mqtt-user und Passwort vergeben. 
   Nun gehts zum OpenDTU. Dort unter MQTT die gleichen EInstellungen wie bei Schritt 4. Unter Baistopic habe ich solar/ entfernt und für meinen Hoymiles hms800/ gewählt. War auch irgendwo ein Tipp. 
   Jetzt OpenDTU und HA vorsichtshalber neu starten und mindestens 5 Minuten warten. 
   
  Du könntest dir noch unter HA ->Addons MQTT Explorer installieren. Dort stellst du eine neue Verbindung her mit der Netzwerkadresse des Pi und die anderen Daten wie unter 4. Jetzt Connect drücken und wenn alles geklappt hat, dann sollte dort dein Basistopic von 5. auftauchen. 
  Ich habe Node-Red installiert und vermutlich deswegen Probleme mit meiner Verbindung gehabt. Falls du das auch hast... In Node-Red eine MQTT Node in den Workflow ziehen, doppelklicken und dort unter Server -> MQTT bearbeiten (Bleistiftsymbol) -> Server die Adresse deines Pi, jetzt Reiter Sicherheit wählen und dort Benutzername und Passwort eingeben. Jetzt auf aktualisieren klicken. Rechts oben auf Übernahme/Deploy und Bestätigen. Nun vorsichtshalber HA neu starten und wieder warten... 
  Viel Erfolg
